### PR TITLE
Added tests for sample

### PR DESCRIPTION
**Pull request by ryanjbaxter**

### CODE REVIEW

File: complete/bookstore/src/test/java/hello/BookstoreApplicationTests.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+@SpringBootTest(webEnvironment= SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class BookstoreApplicationTests {
+
+    @Autowired
+    private TestRestTemplate rest;
+
+    @Test
+    public void recommendedTest() {
+        String resp = rest.getForObject("/recommended", String.class);
+        assertEquals("Spring in Action (Manning), Cloud Native Java (O'Reilly), Learning Spring Boot (Packt)", resp);
```
Comment (marcingrzejszczak) [inline complete/bookstore/src/test/java/hello/BookstoreApplicationTests.java]: Isn't our new normal the usage of BDDAssertions from AssertJ ?

File: complete/reading/src/main/java/hello/ReadingApplication.java [hunk 2]
```diff
@@ -15,6 +18,11 @@
   @Autowired
   private BookService bookService;
 
+  @Bean
+  public RestTemplate rest(RestTemplateBuilder builder) {
```
Comment (marcingrzejszczak) [inline complete/reading/src/main/java/hello/ReadingApplication.java]: Why not just return `new RestTemplate()` ?

File: complete/reading/src/test/java/hello/BookServiceTests.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.client.MockRestServiceServer;
+
+import static org.junit.Assert.assertEquals;
+import static org.springframework.test.web.client.match.MockRestRequestMatchers.requestTo;
+import static org.springframework.test.web.client.response.MockRestResponseCreators.withSuccess;
+
+
+@RunWith(SpringRunner.class)
+@RestClientTest(BookService.class)
```
Comment (marcingrzejszczak) [inline complete/reading/src/test/java/hello/BookServiceTests.java]: LOL haven't seen this one before :P

File: complete/reading/src/main/java/hello/ReadingApplication.java (hunk 2 above)
Comment (marcingrzejszczak) [inline complete/reading/src/main/java/hello/ReadingApplication.java]: Ah it's because of `RestClientTest` ?

Comment (ryanjbaxter) [inline complete/reading/src/main/java/hello/ReadingApplication.java]: I was trying to follow along with what our docs say http://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-testing.html#boot-features-testing-spring-boot-applications-testing-autoconfigured-rest-client

Comment (ryanjbaxter) [inline complete/reading/src/main/java/hello/ReadingApplication.java]: also https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-1.4-Release-Notes#resttemplate-builder

File: complete/bookstore/src/test/java/hello/BookstoreApplicationTests.java (hunk 1 above)
Comment (spencergibb) [inline complete/bookstore/src/test/java/hello/BookstoreApplicationTests.java]: Who is "our"?

Comment (ryanjbaxter) [inline complete/bookstore/src/test/java/hello/BookstoreApplicationTests.java]: Marcin clearly ;)

Comment (marcingrzejszczak) [inline complete/bookstore/src/test/java/hello/BookstoreApplicationTests.java]: Our == "Spring Boot" etc. ;)

Comment (gregturn) [inline complete/bookstore/src/test/java/hello/BookstoreApplicationTests.java]: Ahh, I love this one.

**Pull request merged**